acceptance/tests/external_facts/root_uses_default_external_fact_dir.rb

```diff
@@ -1,95 +1,34 @@
-  end
-
-  step "Agent #{agent}: setup default external facts directory (facts.d)"
-  on(agent, "mkdir -p '#{factsd}'")
-
-  step "Agent #{agent}: create an executable external fact in default facts.d"
-  ext_fact_1 = File.join(factsd, "external_fact_1#{ext}")
-  create_remote_file(agent, ext_fact_1, content_1)
-  on(agent, "chmod +x '#{ext_fact_1}'")
-
-  step "Agent #{agent}: the external fact should resolve"
-  assert_equal("foo", fact_on(agent, "external_fact_1"))
-
-  step "Ensure that Facter honors default facts.d dir in addition to user specified directories"
-  on(agent, "mkdir -p '#{custom_external_dir}'")
-  ext_fact_2 = File.join(custom_external_dir, "external_fact_2#{ext}")
-  create_remote_file(agent, ext_fact_2, content_2)
-  on(agent, "chmod +x '#{ext_fact_2}'")
-
-  step "Agent #{agent}: both external facts should resolve"
-  on(agent, facter("--external-dir '#{custom_external_dir}' external_fact_1 external_fact_2"))
-  assert_match(/external_fact_1 => foo/, stdout)
-  assert_match(/external_fact_2 => bar/, stdout)
-
-  step "Ensure that facts in the default external fact dir are overridden by facts in specified dirs"
-  ext_fact_3 = File.join(custom_external_dir, "external_fact_3#{ext}")
-  create_remote_file(agent, ext_fact_3, override_content)
-  on(agent, "chmod +x '#{ext_fact_3}'")
-
-  step "Agent #{agent}: the fact value from the custom external dir should override that of facts.d"
-  on(agent, facter("--external-dir '#{custom_external_dir}' external_fact_1"))
-  assert_match(/baz/, stdout)
-
-  teardown do
-    on(agent, "rm -f '#{ext_fact_1}' '#{ext_fact_2}' '#{ext_fact_3}'")
+# facter resolves facts from the default facts.d directory
+#      on Unix this can be 3 directories (See fact_directory_precedence.rb)
+# Unix - /opt/puppetlabs/facter/facts.d/
+# Windows - C:\ProgramData\PuppetLabs\facter\facts.d\
+test_name "C87571: facter resolves facts in the default facts.d directory" do
+  tag 'risk:high'
+
+  require 'facter/acceptance/user_fact_utils'
+  extend Facter::Acceptance::UserFactUtils
+
+  agents.each do |agent|
+    os_version = on(agent, facter('kernelmajversion')).stdout.chomp.to_f
+    ext = get_external_fact_script_extension(agent['platform'])
+    factsd = get_factsd_dir(agent['platform'], os_version)
+    fact_file = File.join(factsd, "external_fact_1#{ext}")
+    content = external_fact_content(agent['platform'], 'external_fact', 'external_value')
+
+    teardown do
+      on(agent, "rm -f '#{fact_file}'")
+    end
+
+    step "Agent #{agent}: setup default external facts directory and fact" do
+      on(agent, "mkdir -p '#{factsd}'")
+      create_remote_file(agent, fact_file, content)
+      on(agent, "chmod +x '#{fact_file}'")
+    end
+
+    step "agent #{agent}: resolve the external fact" do
+      on(agent, facter('external_fact')) do |facter_output|
+        assert_equal('external_value', facter_output.stdout.chomp, 'Expected to resolve the external_fact')
+      end
+    end
   end
-end
+end
```

Comment:
Yeah, John was complaining when there too many, so I was removing them, but in RubyMine that means you can end up without 1 at the end of the file if your not careful. I'll fixed it